### PR TITLE
Release 1.0.0 with TCP Support and Dynamic Types

### PR DESCRIPTION
**Pull request by JuanCarlos-Arce**
[Refs #3571](https://eprosima.easyredmine.com/issues/3571) Implementation of the Release 1.0.0 with the new features implemented in Fast-RTPS including

- TCP Transports
- Dynamic Types support

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,97 @@
...
+    project("integration_services" C CXX)
+    set(PROJECT_VERSION_MAJOR "${LIB_MAJOR_VERSION}")
+    set(PROJECT_VERSION_MINOR "${LIB_MINOR_VERSION}")
+    set(PROJECT_VERSION_PATCH "${LIB_PATCH_VERSION}")
+    set(PROJECT_VERSION "${LIB_VERSION_STR}")
+else()
+    cmake_policy(SET CMP0048 NEW)
+    project("integration_services" VERSION "${LIB_VERSION_STR}" LANGUAGES C CXX)
+endif()
+#set(PROJECT_MAJOR_VERSION 0)
```
Comment (richiware) [inline CMakeLists.txt]: Remove these obsoletes comments.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,24 +1,97 @@
...
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -W -Wall -Wextra -pedantic")
     if(SUPPORTS_CXX11)
         add_compile_options(--std=c++11)
     else()
         message(FATAL_ERROR "Compiler doesn't support C++11")
     endif()
 endif()
 
-find_package(fastrtps REQUIRED)
+include(${PROJECT_SOURCE_DIR}/cmake/common/eprosima_libraries.cmake)
```
Comment (richiware) [inline CMakeLists.txt]: You are including this cmake file, but you're not using `eprosima_find_package` instead `find_package` for fastcdr and fastrtps.

File: CMakeLists.txt [hunk 3]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
 set(INCLUDE_INSTALL_DIR include/ CACHE PATH "Installation directory for C++ headers")
+set(INCLUDE_SOURCE_DIR src/ CACHE PATH "Installation directory for C++ sources")
 set(LIB_INSTALL_DIR lib/ CACHE PATH "Installation directory for libraries")
 set(DATA_INSTALL_DIR share/ CACHE PATH "Installation directory for data")
-if(WIN32)
-    set(LICENSE_INSTALL_DIR . CACHE PATH "Installation directory for licenses")
-else()
-    set(LICENSE_INSTALL_DIR ${DATA_INSTALL_DIR}/${PROJECT_NAME} CACHE PATH "Installation directory for licenses")
-endif()
+#set(LICENSE_INSTALL_DIR ./ CACHE PATH "Installation directory for licenses")
```
Comment (richiware) [inline CMakeLists.txt]: where is the license installed?

File: CMakeLists.txt [hunk 4]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
 if(WIN32)
-	add_library(islib
-	    src/dynamicload/dynamicload.cpp
-	    src/ISManager.cpp
-	    src/GenericPubSubTypes.cpp
-	    thirdparty/tinyxml2/tinyxml2.cpp thirdparty/tinyxml2/tinyxml2.h)
+    add_library(${SHAREDLIB_NAME} ${${PROJECT_NAME}_source_files})
 
-	target_include_directories(islib PUBLIC thirdparty/tinyxml2)
+    include_directories("${fastrtps_INCLUDE_DIR}")
```
Comment (richiware) [inline CMakeLists.txt]: this is not necessary. including targets fastcdr and fastrtps in `target_link_libraries` already includes the include directory.

File: CMakeLists.txt [hunk 5]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
+        fastcdr
+        ${CMAKE_DL_LIBS}
+        IPHLPAPI
+        shlwapi)
 else()
-	add_library(islib SHARED
-	    src/dynamicload/dynamicload.cpp
-	    src/ISManager.cpp
-	    src/GenericPubSubTypes.cpp)
+    add_library(${SHAREDLIB_NAME} SHARED ${${PROJECT_NAME}_source_files})
```
Comment (richiware) [inline CMakeLists.txt]: Join win32 and unix condition. Use for this cmake generator expressions. You can see an example on fastrtps develop branch.

File: CMakeLists.txt [hunk 6]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
+# Check MSVC architecture
+###############################################################################
+if(MSVC OR MSVC_IDE)
+    check_msvc_arch()
 endif()
 
-target_link_libraries(isrtpsbridgelib islib)
+# Create libraries
+if(MSVC OR MSVC_IDE)
+    if(EPROSIMA_INSTALLER)
```
Comment (richiware) [inline CMakeLists.txt]: When the installation is being built, the library doesn't have to be compiled. This is not the case. See fastrtps cmakelists.txt to do the same.

File: CMakeLists.txt [hunk 7]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
-target_link_libraries(${PROJECT_NAME} islib)
+        if("${CMAKE_SYSTEM_NAME}" STREQUAL "WindowsStore")
+            set_target_properties(${SHAREDLIB_NAME} PROPERTIES VS_WINRT_COMPONENT "true")
+        endif()
+    endif()
+endif()
 
 install(
     FILES
+        src/xmlUtils.h
```
Comment (richiware) [inline CMakeLists.txt]: Include headers to be used for users should be in a `include` path like other products.

File: CMakeLists.txt [hunk 8]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
-    COMPONENT libraries
-    DESTINATION ${LIB_INSTALL_DIR})
 
-export(
-    TARGETS islib
-    FILE ${BIN_INSTALL_DIR}cmake/config/${PROJECT_NAME}Targets.cmake)
+if(EPROSIMA_INSTALLER)
+    install(
+        FILES
+            src/xmlUtils.h
```
Comment (richiware) [inline CMakeLists.txt]: This installation is already done upper.

File: CMakeLists.txt [hunk 9]
```diff
@@ -29,96 +102,257 @@ IF("${isSystemDir}" STREQUAL "-1")
...
+
+    install(
+        FILES
+            thirdparty/tinyxml2/tinyxml2.h
+        COMPONENT sources
+        DESTINATION ${INCLUDE_SOURCE_DIR}/thirdparty/tinyxml2)
+
+    install(
+        FILES
+            src/main.cpp
```
Comment (richiware) [inline CMakeLists.txt:300]: Why the source code is installed?

File: cmake/modules/Findandroid-ifaddrs.cmake [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# ANDROID_IFADDRS_FOUND
```
Comment (richiware) [inline cmake/modules/Findandroid-ifaddrs.cmake]: Did you use this file?

**Review by richiware — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (JuanCarlos-Arce) [reply CMakeLists.txt]: Removed

File: CMakeLists.txt (hunk 2 above)
Comment (LuisGP) [reply CMakeLists.txt]: Now using `eprosima_find_package`.

File: CMakeLists.txt (hunk 3 above)
Comment (LuisGP) [reply CMakeLists.txt]: Modified to be like Fast-RTPS.

File: CMakeLists.txt (hunk 4 above)
Comment (LuisGP) [reply CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 5 above)
Comment (LuisGP) [reply CMakeLists.txt]: Joined.

File: CMakeLists.txt (hunk 6 above)
Comment (LuisGP) [reply CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 9 above)
Comment (LuisGP) [reply CMakeLists.txt:300]: To generate autotools package on linux. Improved section.

File: cmake/modules/Findandroid-ifaddrs.cmake (hunk 10 above)
Comment (LuisGP) [reply cmake/modules/Findandroid-ifaddrs.cmake]: Nope, removed.

**Review by richiware — APPROVED** [review]

**Pull request merged**